ipatests: Test to check sosreport collects healthcheck.log file

menonsudhir opened the pull request:
This test creates healthcheck.log file in /var/log/ipa/healthcheck/  directory if its not present and then checks that when sosreport command is run it collects the healthcheck log file by checking the console log

mrizwan93 commented on ipatests/test_integration/test_ipahealthcheck.py:
```diff
@@ -1157,6 +1161,42 @@ def test_ds_configcheck_passwordstorage(self, modify_pwdstoragescheme):
+                "--build",
+            ]
+        )
+        assert cmd.returncode == 0
```
This can be removed as `raiseonerr=True` is set by default to `run_command()`

menonsudhir replied:
@mrizwan93 thank you for the review. Have removed the line

mrizwan93 reviewed (approved):
LGTM

flo-renaud reviewed (commented):
Hi @menonsudhir 
2 comments:
- You can add a test ensuring that sos report completes successfully even if there is no file to collect, and even if ipa-healthcheck is not installed
- why did you manually create an empty file instead of calling ipa-healthcheck --output-file /var/log/ipa/healthcheck/healthcheck.log ?

flo-renaud commented on ipatests/test_integration/test_ipahealthcheck.py:
```diff
@@ -1167,6 +1171,76 @@ def test_ds_configcheck_passwordstorage(self, modify_pwdstoragescheme):
+        This fixture creates an empty healthcheck.log file when
+        it is not present in /var/log/ipa/healthcheck/ directory so that
+        sosreport command collects the log file in the report generated
+        by the dry run and then deletes the file once the test is finished.
```
The comment needs to be fixed:
This fixture calls ipa-healthcheck in order to generate /var/log/ipa/healthcheck/healthcheck.log if the file doesn't already exist.
The file is deleted once the test is finished.

flo-renaud commented on ipatests/test_integration/test_ipahealthcheck.py:
```diff
@@ -1167,6 +1171,76 @@ def test_ds_configcheck_passwordstorage(self, modify_pwdstoragescheme):
+                ["ipa-healthcheck", "--output-file", HEALTHCHECK_LOG]
+            )
+        yield
+        self.master.run_command(["rm", HEALTHCHECK_LOG])
```
Add "-f" option to make sure the file is removed without prompting for confirmation.

menonsudhir commented:
> Hi @menonsudhir
> 2 comments:
> 
>     * You can add a test ensuring that sos report completes successfully even if there is no file to collect, and even if ipa-healthcheck is not installed
> 
>     * why did you manually create an empty file instead of calling ipa-healthcheck --output-file /var/log/ipa/healthcheck/healthcheck.log ?
@flo-renaud please have a look at the below changes.
added a testcase to check sosreport completes successfully even without healthcheck log file 
empty file has not created and replaced with creating healthcheck.log file.

flo-renaud commented on ipatests/test_integration/test_ipahealthcheck.py:
```diff
@@ -1167,6 +1171,76 @@ def test_ds_configcheck_passwordstorage(self, modify_pwdstoragescheme):
+        server and then installs.
+        """
+        tasks.uninstall_packages(self.master, HEALTHCHECK_PKG)
+        self.master.run_command(["rm", HEALTHCHECK_LOG])
```
Add "-f" option

flo-renaud commented on ipatests/test_integration/test_ipahealthcheck.py:
```diff
@@ -1167,6 +1171,76 @@ def test_ds_configcheck_passwordstorage(self, modify_pwdstoragescheme):
+        even if there is no healthcheck log file to collect
+        """
+        caseid = "123456"
+        if not self.master.transport.file_exists(HEALTHCHECK_LOG):
```
the "if not self.master.transport.file_exists(HEALTHCHECK_LOG):" is not needed since the remove_healthcheck fixture was called.

flo-renaud commented on ipatests/test_integration/test_ipahealthcheck.py:
```diff
@@ -1167,6 +1171,76 @@ def test_ds_configcheck_passwordstorage(self, modify_pwdstoragescheme):
+                ],
+                raiseonerr=False,
+            )
+        assert cmd.returncode == 0
```
If you remove raiseonerr=False, you can also remove the assert (the test would raise an Exception if the command fails).

flo-renaud commented on ipatests/prci_definitions/temp_commit.yaml:
```diff
@@ -68,7 +68,7 @@ jobs:
       args:
         build_url: '{fedora-latest/build_url}'
-        test_suite: test_integration/test_REPLACEME.py
+        test_suite: test_integration/test_ipahealthcheck.py::TestIpaHealthCheck::test_sosreport_includes_healthcheck
```
Please run the temp commit with `test_integration/test_ipahealthcheck.py::TestIpaHealthCheck`. Otherwise potential interactions are not detected, and the new test `test_sosreport_without_healthcheck_installed` is not executed.

flo-renaud commented on ipatests/test_integration/test_ipahealthcheck.py:
```diff
@@ -1167,6 +1171,76 @@ def test_ds_configcheck_passwordstorage(self, modify_pwdstoragescheme):
+        it is not present in /var/log/ipa/healthcheck/ directory so that
+        sosreport command collects the log file in the report generated
+        by the dry run and then deletes the file once the test is finished.
+        """
```
The comment needs to be updated.
```
This fixture calls ipa-healthcheck in order to create /var/log/ipa/healthcheck/healthcheck.log if the file doesn't already exist, and deletes the file once the test is finished.
```

menonsudhir replied:
Comment was updated.

flo-renaud commented on ipatests/test_integration/test_ipahealthcheck.py:
```diff
@@ -1167,6 +1171,76 @@ def test_ds_configcheck_passwordstorage(self, modify_pwdstoragescheme):
+                ["ipa-healthcheck", "--output-file", HEALTHCHECK_LOG]
+            )
+        yield
+        self.master.run_command(["rm", HEALTHCHECK_LOG])
```
Use `rm -f` to make sure the file is removed.

menonsudhir replied:
Added -f option to rm command.

flo-renaud commented on ipatests/test_integration/test_ipahealthcheck.py:
```diff
@@ -1167,6 +1171,76 @@ def test_ds_configcheck_passwordstorage(self, modify_pwdstoragescheme):
+        """
+        This fixture uninstalls healthcheck package on IPA
+        server and then installs.
+        """
```
The comment needs to be updated: `and deletes /var/log/ipa/healthcheck/healthcheck.log`.

menonsudhir replied:
updated the comment.

flo-renaud commented on ipatests/test_integration/test_ipahealthcheck.py:
```diff
@@ -1167,6 +1171,76 @@ def test_ds_configcheck_passwordstorage(self, modify_pwdstoragescheme):
+                ],
+                raiseonerr=False,
+            )
+        assert cmd.returncode == 0
```
You can also remove raiseonerr=False and remove the assert (by default raiseonerr=True and any failing command raises an Exception).

menonsudhir replied:
Removed raiseonerr=False and also the assert statement

flo-renaud commented on ipatests/test_integration/test_ipahealthcheck.py:
```diff
@@ -1167,6 +1171,76 @@ def test_ds_configcheck_passwordstorage(self, modify_pwdstoragescheme):
+        even if there is no healthcheck log file to collect
+        """
+        caseid = "123456"
+        if not self.master.transport.file_exists(HEALTHCHECK_LOG):
```
The `if not self.master.transport.file_exists(HEALTHCHECK_LOG):` is not needed since the remove_healthcheck fixture was called.

menonsudhir replied:
removed the if statement.

flo-renaud reviewed (commented):
Hi @menonsudhir 
please find my inline comments.

flo-renaud commented:
For future ref: temp commit [result](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/7fdc5a52-7cbf-11eb-95c0-fa163e132092/report.html). The failure is in the teardown method, known issue [8506](https://pagure.io/freeipa/issue/8506).

@menonsudhir you can remove the temp commit.

flo-renaud commented:
master:

* 735355ad03185822da08066ff10667a6e7bb7aa6 ipatests: Test to check sosreport collects healthcheck.log file
